### PR TITLE
Implement drawing an image from a CSS style value into a canvas

### LINKED ISSUE #17432: Paint worklets: Converting a URL to a CanvasImageSource?
```
$ ./mach run -d http://localhost:8000/css-paint-api/paint2d-image.html 
Loading blob blob:http://localhost:8000/43f4fe21df594cfe94e3413ea355e78b
Loading blob blob:http://localhost:8000/43f4fe21df594cfe94e3413ea355e78b
Loading blob blob:http://localhost:8000/43f4fe21df594cfe94e3413ea355e78b
TypeError: argument could not be converted to any of: HTMLImageElement, HTMLCanvasElement, CanvasRenderingContext2D
```
Paint worklets do not have access to HTML elements, so it's not obvious how to convert a URL to an argument to `drawImage`.

Spec issue: https://github.com/w3c/css-houdini-drafts/issues/416

### PR DESCRIPTION
**Pull request by asajeffrey**
Implemented drawing a CSS style value into a canvas, which is needed for the Houdini CSS Paint API.

This PR is dependent on #17364.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #17432.
- [X] These changes do not require tests because the existing CSS paint API wpt test cases test this behaviour.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17634)

### CODE REVIEW

File: components/style/servo/restyle_damage.rs
Comment (emilio) [inline components/style/servo/restyle_damage.rs]: I think this could be too overkill... Could we do better? I guess this is fine for now.

File: components/script/dom/worklet.rs
Comment (emilio) [inline components/script/dom/worklet.rs]: All these members would benefit from some documentation.

**Review by emilio — COMMENTED** [review]
Looks reasonable to me... Probably @jdm should take a look for the DOM bits.

File: components/style/servo/restyle_damage.rs
Comment (asajeffrey) [inline components/style/servo/restyle_damage.rs]: Not sure how much better we could do, rather annoyingly. We could try to be smart about which properties are being monitored by a paint worklet.

File: components/script/dom/worklet.rs
Comment (asajeffrey) [inline components/script/dom/worklet.rs]: Not sure what would be added here, other than "the sender to the primary worklet thread" :)

**Comment by asajeffrey** [comment]
The dependent PRs for this one have landed, so this one is ready for review. @emilio suggested I r? @jdm

**Comment by asajeffrey** [comment]
If #17625 is addressed before this lands, e.g. by @brainlessdeveloper in #17812, we can remove `base_url` from https://github.com/asajeffrey/servo/blob/bf8bcd9a401586d6f0d2e258831e6181d1aaa174/components/script/dom/canvasrenderingcontext2d.rs#L77. URLs should be resolved by style relative to the stylesheet, not by script.

File: components/script/dom/canvasrenderingcontext2d.rs
Comment (jdm) [inline components/script/dom/canvasrenderingcontext2d.rs]: Use /// instead //.

**Review by jdm — APPROVED** [review]

File: components/script/dom/canvasrenderingcontext2d.rs
Comment (asajeffrey) [inline components/script/dom/canvasrenderingcontext2d.rs]: Done.

**Pull request merged**
